pkgs/development/coq-modules/metacoq/default.nix

```diff
@@ -0,0 +1,76 @@
+{ lib, which, fetchzip,
+  mkCoqDerivation, recurseIntoAttrs,  single ? false,
+  coqPackages, coq, equations, version ? null }@args:
+with builtins // lib;
+let
+  repo  = "metacoq";
+  owner = "MetaCoq";
+  defaultVersion = with versions; switch coq.coq-version [
+      { case = "8.11"; out = "1.0-beta2-8.11"; }
+      { case = "8.12"; out = "1.0-beta2-8.12"; }
+      # Do not provide 8.13 because it does not compile with equations 1.3 provided by default (only 1.2.3)
+      # { case = "8.13"; out = "1.0-beta2-8.13"; }
```

Comment:
Your switch could test not just the Coq version but also the Equations version so that MetaCoq is made available as soon as Equations is set at the right version. You could inspire of packages testing both Coq and MathComp version.

Reply:
If I remember correctly @CohenCyril argued orally against this solution mentioning that the version for equations wouldn't be preserved/mentioned in the `.vo` artifacts (and would lead to inconsistent build when importing such artifacts in another system)